src/main/kotlin/com/hibob/academy/resource/PetResource.kt

```diff
@@ -1,78 +1,64 @@
 package com.hibob.academy.resource
 
-import com.hibob.academy.dao.PetData
+import com.hibob.academy.dao.Pet
 import com.hibob.academy.dao.PetType
+import com.hibob.academy.service.PetService
 import jakarta.ws.rs.*
 import jakarta.ws.rs.core.MediaType
 import jakarta.ws.rs.core.Response
 import org.springframework.stereotype.Controller
 import org.springframework.web.bind.annotation.RequestBody
-import java.util.*
 
-class PetResource {
-
-
-    // Get pets by owner: Given an owner ID, return all the information of the pets this owner adopted.
-    // Count pets by type: This API does not receive any information but return a map of the type to how many pets of that type we have in the DB
-
-    @Controller
+@Controller
     @Path("/api/noy/pets")
     @Produces(MediaType.APPLICATION_JSON)
-    class PetsResource {
+    class PetsResource(private val petService: PetService) {
 
-        // create
         @POST
         @Consumes(MediaType.APPLICATION_JSON)
-        fun addPet(@RequestBody pet: PetData): Response {
-            Response.status(Response.Status.CREATED).build()
-            return Response.ok(pet).build()
+        fun addPet(@RequestBody pet: Pet): Response {
+            val petId = petService.createPet(pet)
+            return Response.status(Response.Status.CREATED).entity("new pet created with id $petId").build()
         }
 
-        // update
         @PUT
         @Consumes(MediaType.APPLICATION_JSON)
         @Path("/{petId}/updatePetName")
-        fun updateName(@PathParam("petId") petId: Long, @QueryParam("newName") newName: String?): Response {
-            val pet = PetData(id = petId, name = "Angie", type = PetType.DOG, companyId = 12L, dateOfArrival = Date(), ownerId = null)
-            pet.name = newName ?: pet.name
-            Response.status(Response.Status.ACCEPTED).build()
-            return Response.ok(pet).build()
+        fun updateName(@PathParam("petId") petId: Long, @QueryParam("newName") newName: String, @QueryParam("companyId") companyId: Long): Response {
+            petService.updatePetName(petId, newName, companyId)
+            return Response.status(Response.Status.ACCEPTED).entity("Pets name changed successfully").build()
         }
 
-        // http://localhost:8080/api/noy/pets/updatePetName/3983f705-e3c4-4515-b46e-b8c74936ffd9?newName=Nessy
-
+        @PUT
+        @Consumes(MediaType.APPLICATION_JSON)
+        @Path("/{petId}/{ownerId}/updatePetsOwner")
+        fun updateOwnerForPet(@PathParam("petId") petId: Long, @PathParam("ownerId") ownerId: Long, @QueryParam("companyId") companyId: Long): Response {
+            val success = petService.updatePetOwner(petId, ownerId, companyId)
+            if(success == 1)
+                return Response.status(Response.Status.OK).entity("Pet $petId got a new owner!").build()
+            return Response.status(Response.Status.BAD_REQUEST).entity("Pet doesn't exist or pet $petId already has an owner!").build()
```

Comment:
Why not handle the success in the service and throw an exception if doesnt success?

Reply:
fixed it